packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/404.tsx

```diff
@@ -0,0 +1,46 @@
+import { sitecorePagePropsFactory } from 'lib/page-props-factory';
+import Layout from 'src/Layout';
+import config from 'temp/config';
+import {
+  ComponentPropsContext,
+  GraphQLErrorPagesService,
+  SitecoreContext,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import { componentFactory } from 'temp/componentFactory';
+import { SitecorePageProps } from 'lib/page-props';
+import NotFound from 'src/NotFound';
+
+const Custom404 = (props: SitecorePageProps): JSX.Element => {
+  if (!(props && props.layoutData)) {
+    return <NotFound />;
+  }
+
+  return (
+    <ComponentPropsContext value={{}}>
+      <SitecoreContext componentFactory={componentFactory} layoutData={props.layoutData}>
+        <Layout layoutData={props.layoutData} headLinks={props.headLinks} />
+      </SitecoreContext>
+    </ComponentPropsContext>
+  );
+};
+
+export async function getStaticProps(context: SitecorePageProps) {
+  const props = await sitecorePagePropsFactory.create(context);
+  const errorPagesService = new GraphQLErrorPagesService({
+    endpoint: config.graphQLEndpoint,
+    apiKey: config.sitecoreApiKey,
+    siteName: props.site.name,
```

Comment:
Do we have a problem supporting multisite? Custom static 404 and 500 pages are generated at build time and AFAIK, there will only be 1 version of each.

Reply:
I didn't check it with multisite because this is hotfix for our customers who uses single site version. But I will create PBI for check and implement functionality for multisite version